Tests/L2Tests/L2TestsPlugin/tests/Warehouse_L2Test.cpp

```diff
@@ -0,0 +1,782 @@
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_FALSE(result["success"].Boolean());
+    EXPECT_FALSE(result["clean"].Boolean());
+    EXPECT_STREQ("[]", result["files"].String().c_str());
+
+    //Empty conf file
+    Core::File fileConf(customDataFile);
+    Core::Directory(fileConf.PathName().c_str()).CreatePath();
+    fileConf.Create();
+   
+    params["age"]=300;
+    status = InvokeServiceMethod("org.rdk.Warehouse.1", "isClean", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_FALSE(result["success"].Boolean());
+    EXPECT_FALSE(result["clean"].Boolean());
+    EXPECT_STREQ("[]", result["files"].String().c_str());
+
+    //Create empty conf file
+    fileConf.Write(customDataFileContent, sizeof(customDataFileContent));
+
+    params["age"]=300;
+    status = InvokeServiceMethod("org.rdk.Warehouse.1", "isClean", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_TRUE(result["success"].Boolean());
+    EXPECT_TRUE(result["clean"].Boolean());
+    EXPECT_STREQ("[]", result["files"].String().c_str());
+   
+    //Add test data to conf file
+    Core::File filePref(userPrefFile);
+    Core::Directory(filePref.PathName().c_str()).CreatePath();
+    filePref.Create();
+    filePref.Open();
+    filePref.Write(userPrefLang, sizeof(userPrefLang));
+    
+    params["age"]=-4;
+    status = InvokeServiceMethod("org.rdk.Warehouse.1", "isClean", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_TRUE(result["success"].Boolean());
+    EXPECT_FALSE(result["clean"].Boolean());
+    EXPECT_STREQ("[\"\\/opt\\/user_preferences.conf\"]", result["files"].String().c_str());
+
+    fileConf.Destroy();
+    filePref.Destroy();
+}
+
+/********************************************************
+************Test case Details **************************
+** 1. TEST_F to achieve max. Lcov
+** 2. To enable RFC_PWRMGR2 -- "Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.Power.PwrMgr2.Enable"
+** 2. Triggered resetDevice Method 
+** 3. Verify the response of resetDevice Method for different "resetType"
+*******************************************************/
+TEST_F(Warehouse_L2Test,Warehouse_Lcov_ResetDevice)
+{
+    JSONRPC::LinkType<Core::JSON::IElement> jsonrpc(WAREHOUSE_CALLSIGN,WAREHOUSEL2TEST_CALLSIGN );
+    StrictMock<AsyncHandlerMock_Warehouse > async_handler;
+    uint32_t status = Core::ERROR_GENERAL;
+    JsonObject params;
+    JsonObject result;
+    uint32_t signalled = WAREHOUSEL2TEST_STATE_INVALID;
+    std::string message;
+    bool m_isPwrMgr2RFCEnabled;
+    JsonObject expected_status;
+
+    /* Deactivate plugin in TEST_F*/
+    status = DeactivateService("org.rdk.Warehouse");
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    /* Activate plugin in TEST_F*/
+    status = ActivateService("org.rdk.Warehouse");
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    ON_CALL(*p_rfcApiImplMock, getRFCParameter(::testing::_, ::testing::_, ::testing::_))
```

Comment:
Change this to EXPECT_CALL() and move after DeactivateService() and before ActivateService().